lesson10/main.js

```diff
@@ -0,0 +1,57 @@
+const ul = document.getElementById("js-ul");
+
+const addLoading = () =>{
+  const loading = document.getElementById("js-loading");
+  const loadingImg = document.createElement('img');
+  loadingImg.src = 'loading.gif';
+  loadingImg.id = "loading-img";
+  loading.appendChild(loadingImg);
+};
+
+const removeLoading = () =>{
+  document.getElementById('loading-img').remove();
+};
+
+const createLists = (features) =>{
+  const fragment = new DocumentFragment();
+
+  for (let feature of features ){
+      const li = document.createElement('li');
+      const a = document.createElement('a');
+      const img = document.createElement('img');
+      a.href = feature.to;
+      a.textContent = feature.text;
+      img.src = feature.img;
+      img.alt = feature.alt;
+
+      fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin', img);
+    };
+
+    ul.appendChild(fragment);
+};
+
+const getData = new Promise((resolve) =>{
+  setTimeout(() => resolve( [
+    { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+    { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
+  ]), 3000)
+});
```

Comment:
I think it is possible to write the following.
[CodeSandBox](https://codesandbox.io/s/lesson10-forked-2bxyjq?file=/main.js)
```JavaScript
const getData = new Promise((resolve) => {
  setTimeout(
    () =>
      resolve(
        createLists([
          {to: "bookmark.html",img: "1.png",alt: "画像1",text: "ブックマーク"},
          { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
        ])
      ),
    3000
  );
});
```

Reply:
Thank you for suggestion Mai-san,

I still do not understand very much why the difference between passing argument(features array) inside of the renderLists() function and passing the argument inside of the promise function. 
I appreciate if you could show me why your suggested example code is more acceptable than just passing a function?